airgun/entities/computeresource.py

```diff
@@ -169,9 +242,61 @@ class ComputeResourceVMImport(NavigateStep):
 
     def prerequisite(self, *args, **kwargs):
         entity_name = kwargs.get('entity_name')
-        return self.navigate_to(self.obj, 'Detail', rhev_name=entity_name)
+        return self.navigate_to(self.obj, 'Detail', entity_name=entity_name)
 
     def step(self, *args, **kwargs):
         vm_name = kwargs.get('vm_name')
         self.parent.virtual_machines.search(vm_name)
         self.parent.virtual_machines.actions.fill("Import")
+
+
+@navigator.register(ComputeResourceEntity, 'Create Image')
+class ComputeResourceImageCreate(NavigateStep):
+
+    PROVIDER_VIEWS = dict(
+        RHV=ComputeResourceRHVImageCreateView,
+    )
+
+    def prerequisite(self, *args, **kwargs):
+        entity_name = kwargs.get('entity_name')
+        parent_view = self.navigate_to(self.obj, 'Detail', entity_name=entity_name)
+        provider = parent_view.compute_resource.table.row(property='Provider')['Value'].read()
+        view = self.PROVIDER_VIEWS.get(provider)
+        if not view:
+            raise ValueError('Provider type "{0}" for class "{1}" not implemented'.format(
+                provider, self.__class__.__name__))
+        self.VIEW = view
+        return parent_view
+
+    def step(self, *args, **kwargs):
+        self.parent.create_image.click()
+
+    def am_i_here(self, *args, **kwargs):
+        return self.view.is_displayed
+
+
+@navigator.register(ComputeResourceEntity, 'Edit Image')
+class ComputeResourceImageEdit(NavigateStep):
+
+    PROVIDER_VIEWS = dict(
+        RHV=ComputeResourceRHVImageEditView,
+    )
+
+    def prerequisite(self, *args, **kwargs):
```

Comment:
```
```
entity_name = kwargs.get('entity_name')
parent_view = self.navigate_to(self.obj, 'Detail', entity_name=entity_name)
self.VIEW = ComputeResourceRHVImageEditView
return parent_view
```

Reply:
I do not agree with this change.